pkgs/tools/misc/coreutils/default.nix

```diff
@@ -42,6 +42,12 @@ stdenv.mkDerivation rec {
   patches = lib.optionals stdenv.hostPlatform.isMusl [
     # https://lists.gnu.org/archive/html/bug-coreutils/2024-03/msg00089.html
     ./fix-test-failure-musl.patch
+  ] ++ [
+    # https://lists.gnu.org/archive/html/bug-coreutils/2024-05/msg00037.html
+    # This is not precisely the patch provided - this is a diff of the Makefile.in
+    # after the patch was applied and autoreconf was run, since adding autoreconf
+    # here causes infinite recursion.
+    ./fix-mix-flags-deps-libintl.patch
   ];
```

Comment:
```suggestion
  patches = [
    # https://lists.gnu.org/archive/html/bug-coreutils/2024-05/msg00037.html
    # This is not precisely the patch provided - this is a diff of the Makefile.in
    # after the patch was applied and autoreconf was run, since adding autoreconf
    # here causes infinite recursion.
    ./fix-mix-flags-deps-libintl.patch
  ] ++ lib.optionals stdenv.hostPlatform.isMusl [
    # https://lists.gnu.org/archive/html/bug-coreutils/2024-03/msg00089.html
    ./fix-test-failure-musl.patch
  ];
```